sinkingship.py

```diff
@@ -169,12 +188,23 @@ def save_post_media_files(info_path):
     post_dir = os.path.dirname(info_path)
     post_id = post_data["id"]
 
-    if post_data["type"] == "photo":
+    if post_data["type"] in ("photo", "link"):
+        if not "photos" in post_data:
+            return
```

Comment:
minor: Python convention is “easier to seek forgiveness than ask permission”, not “look before you leap” – meaning, we should try the code, and catch the error if it fails.

In this case, if you're seeing photo posts without a `"photos"` block, we should wrap this in a `try … except KeyError:` (and provide a comment pointing to the post in question!).

Unless it's caused by putting the `"link"` type here? I'm not sure why you've done that.